saleor/order/error_codes.py

```diff
@@ -36,6 +36,8 @@ class OrderErrorCode(Enum):
     CHANNEL_INACTIVE = "channel_inactive"
     INVALID_VOUCHER = "invalid_voucher"
     INVALID_VOUCHER_CODE = "invalid_voucher_code"
+    NON_EDITABLE_GIFT_LINE_QUANTITY = "non_editable_gift_line_quantity"
```

Comment:
Wouldn't `NON_EDITABLE` and `NON_REMOVABLE` be better?
If you place `GIFT_LINE` here, this error will be related only to giftcards (it is ok for now, but if we extend saleor in the future, with functionality, where you can't edit the line for some reason (like bundle products etc), this error will not work).

---

Also don't think that `_QUNATITY` is useful here. Like the error that storefront receives will be:
```python
{
                     "quantity": ValidationError(
                         "Quantity of order line marked as gift can't be edited.",
                         code=CheckoutErrorCode.NON_EDITABLE_GIFT_LINE_QUANTITY.value,
                         params={"lines": global_ids},
                     )
                 }
```
You already have the error assigned to field `quantity`, you also have `quantity` in error message. 
The most important info for the storefront here is that the line is non editable/ non removable.

Reply:
I deleted `_QUANTITY`, but kept `GIFT_LINE` part as we don't have `gift_line` field. So to determine the reason the line is not editable, would require additional check, like `line.is_gift=True`